internal/handlers/common/aggregations/operators/operators.go

```diff
@@ -13,27 +13,96 @@
 // limitations under the License.
 
 // Package operators provides aggregation operators.
+// Operators are used in aggregation stages to filter and model data.
+// This package contains all operators apart from the accumulation operators,
+// which are stored and described in accumulators package.
+//
+// Accumulators that can be used outside of accumulation with different behaviour (like `$sum`),
+// should be stored in both operators and accumulators packages.
 package operators
 
-import (
-	"context"
+import "github.com/FerretDB/FerretDB/internal/types"
+
+// newOperatorFunc is a type for a function that creates a standard aggregation operator.
+//
+// By standard aggregation operator we mean any operator that is not accumulator.
+// While accumulators perform operations on multiple documents
+// (for example `$count` can count documents in each `$group` group),
+// standard operators perfrom operations on single documents.
+type newOperatorFunc func(expression *types.Document) (Operator, error)
+
+// Operator is a common interface for standard aggregation operators.
+type Operator interface {
+	// Process document and returns the result of applying operator.
+	Process(in *types.Document) (any, error)
+}
+
+//go:generate ../../../../../bin/stringer -linecomment -type OperatorErrorCode
 
-	"github.com/FerretDB/FerretDB/internal/types"
+// OperatorErrorCode represents aggregation operator error code.
+type OperatorErrorCode int
```

Comment:
I know a plenty of people like having everything in the same file, but to me it looks like errors should be in a separate file. As otherwise the structure of this file is not super clear. We have interface definition on top, and then something about errors, and then something to create new operators. This structure looks a bit strange to me.

Reply:
As I removed error codes completely and there's not a lot of code related to errors right now, I moved everything back to the operators.go file, is that ok?